### ISSUE TITLE
The broker reports an error: No such entry 

### ISSUE THREAD

**Issue by lordcheng10** [apache/pulsar#15111]
**Describe the bug**
pulsar version: 2.8.1
bookkeeper config: journalWriteData=false

**error log:**
10:55:04.879 [BookKeeperClientWorker-OrderedExecutor-12-0] ERROR org.apache.pulsar.broker.service.persistent.PersistentDispatcherMultipleConsumers - [persistent://teg_onion_onion_common_data_nj/teg_onion_onion_common_data_nj/teg_onion_onion_common_data_nj-partition-156 / t_teg_onion_b_teg_onion_onion_common_data_nj_cg_onion_common_data_cos_svr_nj_001] Error reading entries at 9417597:1325183 : No such entry, Read Type Normal - Retrying to read in 54.075 seconds
10:55:11.454 [BookKeeperClientWorker-OrderedExecutor-61-0] ERROR org.apache.bookkeeper.client.PendingReadOp - Read of ledger entry failed: L9417597 E1325183-E1325183, Sent to [9.150.226.98:3181, 9.150.228.32:3181], Heard from [] : bitset = {}, Error = 'No such entry'. First unread entry is (-1, rc = null)

From the log, the two replica data are stored on two bookie nodes: 9.150.226.98:3181, 9.150.228.32:3181
Execute the commands on the two bookie nodes respectively: sh bin/bookkeeper  shell ledger 9417597
<img width="866" alt="image" src="https://user-images.githubusercontent.com/19296967/162671454-c8d41ceb-cae8-4388-9232-b901ba053ac9.png">
<img width="795" alt="image" src="https://user-images.githubusercontent.com/19296967/162671497-92d4f942-390a-4457-a1ab-4511e6a01b68.png">

**Comment by lhotari** [comment]
Would you have a chance to check if the problem can be reproduced with Netty Recycler `-Dio.netty.recycler.maxCapacityPerThread=0` . The reason I'm bringing this up is that in BookKeeper client there's a known issue fixed by https://github.com/apache/bookkeeper/pull/3110 by @congbobo184 that isn't yet available in a released version of BookKeeper client. The workaround for the problem is to disable the Netty Recycler in Pulsar Broker (which uses the BK client)

**Comment by hangc0276** [comment]
It is not caused by bookie client mis fill entry. We can't read the target entry by bin/bookkeeper shell command.

**Comment by lhotari** [comment]
> It is not caused by bookie client mis fill entry. We can't read the target entry by bin/bookkeeper shell command.

true. It would be interesting to see if something gets messed up due to invalid recycling of objects. The quick way to test that is to disable Netty Recycler with `-Dio.netty.recycler.maxCapacityPerThread=0`. The performance is very similar with newer JDKs when disabling Netty Recycler.
